### ISSUE TITLE
Please let's not use "freeze" or "frozen" for this concept.

### ISSUE THREAD

**Issue by erights** [bakkot/proposal-frozen-set#2]
I regret having used the term "freeze" for the `Object.freeze` functionality. It's purpose is to make an object's API surface tamper proof. If has little to do with immutability. Unfortunately, we did go with that word, so that's what it means.

```js
let x = 0;
export const count = Object.freeze(() => ++x);
```

exports a frozen function. Likewise, for

```js
export const s = Object.freeze(new Set());
```

we would *already* say that this exports a frozen set.

Again, my apologies for introducing this terminology confusion in JS, but we must adapt to it.

**Comment by ljharb** [comment]
`immutable-set`?

**Comment by bakkot** [comment]
I agree that this is probably the right call, at least for the API itself. (For the discussion I think it's probably OK; I doubt people are going to be confused.)

Not sure what else to go with, though. I'm hesitant to use "immutable", since to many people that implies the existence of methods for performing immutable updates, i.e., methods like `.add` which instead of mutating return a new data structure reflecting the change. Such data structures are also very useful, but significantly more complex to design and implement.

(... on the other hand, maybe we should be considering them? I'll add something to the readme.)

**Comment by ljharb** [comment]
- immutable
 - locked
 - read-only

**Comment by bradennapier** [comment]
Yeah I think it’s clear that freeze is not the proper word.  

Perhaps a better solution lies in a broader set of functionality that attempts to consider the idea of general immutability within the language? 

Seems most of the disagreements end up being due to the idea not being compatible with other, similar objects? 

Just a thought, it’d be beyond me on what something like that would need to look internally, but you guys can certainly tell me! Heh.

**Comment by bradennapier** [comment]
@erights “Tamper proof” makes me think “locked” immediately, personally.

**Comment by erights** [comment]
Not "read-only" because that already names a necessary concept also introduced by this proposal: a read-only of a possible mutable collection.

To me, "immutable" suggests a transitive property, not a shallow one. An immutable set I would expect to contain only immutable members. Immutable things can't lead to mutable state. This is how the Joe-E language used the term and it seemed very natural.

Sorry, but I don't yet have a positive suggestion. E used `ConstSet`, but that seems strange for JS. How do others react to `ConstSet`?

**Comment by bakkot** [comment]
> How do others react to `ConstSet`?

I react the same as you - it seems strange in JS. Beginners already struggle quite a bit with the idea of `const` meaning only that the reference itself is constant.

**Comment by ljharb** [comment]
`FixedSet`, `FinalizedSet`?

**Comment by erights** [comment]
I ***like*** `FixedSet` !

**Comment by erights** [comment]
That was four days ago (though over a weekend). Is there general comfort with `FixedSet` etc?

**Comment by bakkot** [comment]
`FixedSet` seems good to me. I've added a note to the readme suggesting it.

**Comment by bradennapier** [comment]
Agreed.

**Comment by bakkot** [comment]
@ljharb points out on IRC that `FixedSet` implies that the current `Set` is broken. That sense of "fixed" is definitely a more common use than "fixed" as in "fixed point".

**Comment by erights** [comment]
> @ljharb points out on IRC that FixedSet implies that the current Set is broken.

Damn. Ok, more suggestions?

**Comment by normanzb** [comment]
> Damn. Ok, more suggestions?

`SealedSet`, `ReadOnlySet`, `InalterableSet`, `UnmodifiableSet` or perhaps, don't call it `Set` but call it something else that represent a thing that records information and it lasts a long time and mostly unchangable... such as `Stele`?

**Comment by erights** [comment]
Can we move this issue to https://github.com/tc39/proposal-readonly-collections ?
These suggestions are just as relevant over there, and it is the active successor of this proposal.

**Comment by ljharb** [comment]
@erights if this repo is first transferred to the tc39 org, then this issue can be transferred between the repos
